pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -507,7 +507,7 @@ func (r *ReconcileClusterDeployment) reconcile(request reconcile.Request, cd *hi
 	}
 
 	if cd.DeletionTimestamp != nil {
-		if !controllerutils.HasFinalizer(cd, hivev1.FinalizerDeprovision) {
+		if !controllerutils.HasFinalizer(cd, hivev1.FinalizerDeprovision) && !controllerutils.HasFinalizer(cd, hivev1.FinalizerCustomizationRelease) {
```

Comment:
Why a second finalizer that this one controller is responsible for?
More generally, why is the clusterDeployment controller responsible for cleanup of the ClusterPool-specific details regarding CustomizationRelease? Before this PR, the clusterdeployment controller had zero handling of anything ClusterPool-related (except for some suspicious code in the metrics calculation). The clusterDeployment.spec.clusterPoolRef was basically an area carved out for the clusterPool controller to be able to more easily perform its task.
My gut tells me that this accounting of what CustomizationRelease is in use or isn't should be handled by the clusterPool controller. Is there some specific reason the approach in this PR was chosen?

Reply:
Clusterpool maintains a list customization names in its inventory, so multiple clusterpool could be using the same customization. From my understanding of Finalizers, the deleted resource should handle the process of releasing or deletion of dependend resources. Customization must be released only when a cluster is completely deleted, so it makes sense to me that it will udpate the customization resource rather than clusterpool.

Reply:
I would write a function named `ensureClusterDeploymentCustomizationIsReleased` in `syncDeletedClusterDeployment` and based on output of that clear the finalizer. This is similar to how we have written `ensureClusterDeprovisioned`. Does it make sense?

Reply:
@akhil-rane sounds good to me